test/wrapper_array.jl

```diff
@@ -320,3 +320,9 @@ const cnda = NamedDimsArray([10 20; 30 40], (:x, :y))
     @test @ballocated(cnda[x=1]) == @ballocated(cnda[1, :])
     @test 0 == @ballocated cnda[x=1, y=1] = 66
 end
+
+@testset "convert" begin
+    nda = NamedDimsArray{(:id, :_)}(rand(5,5))
```

Comment:
Might want to test on cases where the eltypes don't match to confirm that it fails correclty?

Reply:
Technically these are being _allowed_ given the discussion in the other thread. But they can be easily changed to test for breaking if that's the choice.